### PR TITLE
Add support for IS v2 API with authentication

### PR DESCRIPTION
**Pull request by jryans**
This adds support for the IS v2 API in all existing IS APIs. It also adds new support for request IS account tokens (which are new in the v2 API version).

This will fall back to v1 APIs if v2 is not supported for the moment, but this will eventually be removed once servers are updated.

Implements part of [MSC2140](https://github.com/matrix-org/matrix-doc/pull/2140)
Fixes part of https://github.com/vector-im/riot-web/issues/10408

### CODE REVIEW

**Comment by turt2live** [comment]
> This will fall back to v1 APIs if v2 is not supported.

for outside readers: the plan is to not continue this fallback once servers are reasonably updated.

**Comment by jryans** [comment]
> > This will fall back to v1 APIs if v2 is not supported.
> 
> for outside readers: the plan is to not continue this fallback once servers are reasonably updated.

Thanks, wasn't quite sure how to state that at the JS SDK level. Updated original comment to match.

File: src/base-apis.js [hunk 1]
```diff
@@ -1717,22 +1739,49 @@ MatrixBaseApis.prototype.getKeyChanges = function(oldToken, newToken) {
  * @param {string} nextLink Optional If specified, the client will be redirected
  *                 to this link after validation.
  * @param {module:client.callback} callback Optional.
+ * @param {string} isAccessToken The `access_token` field of the Identity Server
```
Comment (turt2live) [inline src/base-apis.js]: like the react-sdk: `accessToken` or `identityAccessToken` please. Booleans are not strings.

File: src/base-apis.js [hunk 2]
```diff
@@ -1717,22 +1739,49 @@ MatrixBaseApis.prototype.getKeyChanges = function(oldToken, newToken) {
...
-        callback, "POST", "/validate/email/requestToken",
-        params, httpApi.PREFIX_IDENTITY_V1,
-    );
+
+    try {
+        const response = await this._http.idServerRequest(
+            undefined, "POST", "/validate/email/requestToken",
+            params, httpApi.PREFIX_IDENTITY_V2, isAccessToken,
+        );
+        // TODO: Fold callback into above call once v1 path below is removed
```
Comment (turt2live) [inline src/base-apis.js]: we should also make sure to have a maintenance issue on riot-web to clean this fallback up at a future date. Not a blocker for this PR to be merged, but should be done sometime around the merge at latest.

File: src/base-apis.js (hunk 1 above)
Comment (jryans) [reply src/base-apis.js]: Right, sorry for the naming... All fixed! Still not sure why it sounded okay at the time... 😅

File: src/base-apis.js (hunk 2 above)
Comment (jryans) [reply src/base-apis.js]: Right, good call. I filed https://github.com/vector-im/riot-web/issues/10443 for this. I'll link to it in the code as well.

**Pull request merged**